xyzservices/lib.py

```diff
@@ -28,6 +28,15 @@ def __call__(self, **kwargs):
         new.update(kwargs)
         return new
 
+    def requires_token(self):
+        """
+        Returns True if the TileProvider requires access token to fetch tiles
+        """
+        # both attribute and placeholder in url are required to make it work
+        if hasattr(self, "accessToken") and "accessToken" in self.url:
```

Comment:
I am not sure if we shouldn't add some check to TileProvider itself ensuring that there is `accessToken` attribute if there is a placeholder in URL. Because otherwise the URL will be broken.